operators/Makefile

```diff
@@ -46,6 +46,9 @@ GO_LDFLAGS := -X github.com/elastic/cloud-on-k8s/operators/pkg/about.version=$(V
 	-X github.com/elastic/cloud-on-k8s/operators/pkg/about.buildDate=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ') \
 	-X github.com/elastic/cloud-on-k8s/operators/pkg/about.buildSnapshot=$(SNAPSHOT)
 
+# Setting for CI, if set to true will prevent building and using local Docker image
+USE_CUSTOM_IMAGE ?= false
```

Comment:
`USE_CUSTOM_IMAGE ` does not sound like a very explicit setting to me. We are always using a custom image?

Reply:
I think we are adding some more complexity in the Makefile.
Is there a way we could do things the other way around to reuse as much as possible our existing targets instead of duplicating them into new targets that partially duplicate the behaviour?
 `OPERATOR_IMAGE=my.custom.image/eck SKIP_DOCKER_PUSH=true make ci-e2e`

Reply:
I share the concerns about the code duplication that we add with this.

I'm wondering if it will be simpler to split the current job in 2 jobs: a first that build and push the artefacts (operator docker image, yaml and ci docker image) for each merge in master and a second that runs the e2e tests given a version/sha1?

Reply:
Fixed